src/fileoverview_comment_transformer.ts

```diff
@@ -28,6 +28,43 @@ export function isClosureFileoverviewComment(text: string) {
   return current !== null && current.tags.some(t => FILEOVERVIEW_COMMENT_MARKERS.has(t.tagName));
 }
 
+/**
+ * Given a parsed @fileoverview comment, ensures it has all the attributes we need.
+ * This function can be called to modify an existing comment or to make a new one.
+ *
+ * @param tags Comment as parsed list of tags; modified in-place.
+ */
+function modifyFileoverviewComments(tags: jsdoc.Tag[]) {
+  // Ensure we start with a @fileoverview.
+  if (!tags.find(t => t.tagName === 'fileoverview')) {
+    tags.splice(0, 0, {tagName: 'fileoverview', text: 'added by tsickle'});
+  }
+
+  // Find or create a @suppress tag.
+  // Closure compiler barfs if there's a duplicated @suppress tag in a file, so the tag must
+  // only appear once and be merged.
+  let suppressTag = tags.find(t => t.tagName === 'suppress');
+  let suppressions: Set<string>;
+  if (suppressTag) {
+    suppressions = new Set((suppressTag.type || '').split(',').map(s => s.trim()));
+  } else {
+    suppressTag = {tagName: 'suppress', text: 'checked by tsc'};
+    tags.push(suppressTag);
+    suppressions = new Set();
+  }
+
+  // Ensure our suppressions are included in the @suppress tag:
+  // 1) Suppress checkTypes.  We believe the code has already been type-checked by TypeScript,
+  // and we cannot model all the TypeScript type decisions in Closure syntax.
+  suppressions.add('checkTypes');
+  // 2) Suppress extraRequire.  We remove extra requires at the TypeScript level, so any require
+  // that gets to the JS level is a load-bearing require.
+  suppressions.add('extraRequire');
+  suppressTag.type = Array.from(suppressions.values()).sort().join(',');
```

Comment:
FYI `Set` in JavaScript is a linked hash set, `.values()` iterates in insertion order, so you don't have to sort here (unless you generally want these tags to be emitted in order).

Reply:
The set could have already contain extraRequire (due to an existing suppress tag) before our add, so it would come out first in that case.  I guess it'd still be stable in that case.  Hm, maybe it's better to have our additions to the list always come last?

Reply:
I think it really doesn't matter in either way.